orca-migration/src/main/kotlin/com/netflix/spinnaker/orca/pipeline/persistence/migration/OrchestrationMigrationAgent.kt

```diff
@@ -0,0 +1,86 @@
+
+import com.netflix.spinnaker.orca.ExecutionStatus
+import com.netflix.spinnaker.orca.front50.Front50Service
+import com.netflix.spinnaker.orca.notifications.AbstractPollingNotificationAgent
+import com.netflix.spinnaker.orca.notifications.NotificationClusterLock
+import com.netflix.spinnaker.orca.pipeline.model.Execution
+import com.netflix.spinnaker.orca.pipeline.model.Execution.ExecutionType.ORCHESTRATION
+import com.netflix.spinnaker.orca.pipeline.persistence.DualExecutionRepository
+import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionRepository
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression
+import org.springframework.stereotype.Component
+
+@Component
+@ConditionalOnBean(DualExecutionRepository::class)
+@ConditionalOnExpression("\${pollers.orchestrationMigration.enabled:false}")
+class OrchestrationMigrationAgent(
+  clusterLock: NotificationClusterLock,
+  private val front50Service: Front50Service,
+  private val dualExecutionRepository: DualExecutionRepository,
+  @Value("\${pollers.orchestrationMigrator.intervalMs:3600000}") private val pollingIntervalMs: Long
+) : AbstractPollingNotificationAgent(clusterLock) {
+
+  private val log = LoggerFactory.getLogger(javaClass)
+
+  override fun startPolling() {
+    val previouslyMigratedOrchestrationIds = dualExecutionRepository.primary.retrieveAllExecutionIds(ORCHESTRATION)
+
+    val executionCriteria = ExecutionRepository.ExecutionCriteria().apply {
+      limit = 1000
+      setStatuses(ExecutionStatus.COMPLETED.map { it.name })
+    }
+
+    val allApplications = front50Service.allApplications
+    log.info("Found ${allApplications.size} applications")
+
+    allApplications.forEachIndexed { index, application ->
+      val applicationName = application.name.toLowerCase()
+      val unmigratedOrchestrations = dualExecutionRepository.previous
+        .retrieveOrchestrationsForApplication(applicationName, executionCriteria)
+        .filter { !previouslyMigratedOrchestrationIds.contains(it.id) }
+        .toList()
+        .toBlocking()
+        .single()
+
+      val pendingOrchestrations = mutableListOf<Execution>()
+      val migratableOrchestrations = unmigratedOrchestrations.filterNotTo(pendingOrchestrations) { it.status.isComplete }
+
+      if (pendingOrchestrations.isNotEmpty()) {
+        log.info("${pendingOrchestrations.size} orchestrations yet to complete $applicationName [$index/${allApplications.size}]")
+      }
+
+      if (migratableOrchestrations.isNotEmpty()) {
+        log.info("${migratableOrchestrations.size} orchestrations to migrate ($applicationName) [$index/${allApplications.size}]")
+
+        migratableOrchestrations.forEach {
+          val execution = dualExecutionRepository.previous.retrieve(it.type, it.id)
```

Comment:
This was an oddity of the prior migrators that I kept. I'm not totally certain why we're refreshing the execution even though we already have it, but I'm not eager to deviate from the existing logic. This choice is consistent for both migrators.

Reply:
This was me being stupid, safe to drop and avoid the duplicate read.